corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -253,7 +254,7 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
             this.parentView.setStickyQueryInputs();
         },
         changeDateQueryField: function (e) {
-            this.changeQueryField(e);
+            this.model.set('value', $(e.currentTarget).val());
```

Comment:
This is the bug: the model's value wasn't getting set for date questions (this call to `changeQueryField` does nothing, because `changeQueryField` returns early for date fields).